File: src/main/java/de/tectoast/emolga/commands/draft/RevealPrismaTeamCommand.java
package de.tectoast.emolga.commands.draft;

import de.tectoast.emolga.buttons.buttonsaves.PrismaTeam;
import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.Constants;
import de.tectoast.jsolf.JSONObject;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.interactions.components.buttons.Button;

import java.util.Collections;
import java.util.List;

public class RevealPrismaTeamCommand extends Command {

    public RevealPrismaTeamCommand() {
        super("revealprismateam", "Revealt die Prisma Teams lol", CommandCategory.Draft, Constants.FLPID);
        setCustomPermissions(PermissionPreset.fromIDs(297010892678234114L));
        setArgumentTemplate(ArgumentManagerTemplate.builder()
                .add("user", "User", "Der User lol", ArgumentManagerTemplate.DiscordType.USER)
                .setExample("!revealprismateam @HennyHahn")
                .build());
    }

    @Override
    public void process(GuildCommandEvent e) {
        Member user = e.getArguments().getMember("user");
        JSONObject prisma = getEmolgaJSON().getJSONObject("drafts").getJSONObject("Prisma");
        JSONObject picks = prisma.getJSONObject("picks");
        List<JSONObject> jsonList = picks.getJSONList(user.getId());
        Collections.reverse(jsonList);
        e.getChannel().sendMessage("**" + user.getEffectiveName() + "**").setActionRow(Button.primary("prisma;lol", "Nächstes Pokemon"))
                .queue(m -> prismaTeam.put(m.getIdLong(), new PrismaTeam(jsonList.stream()
                        .map(o -> o.getString("name")).toList(), prisma.getLongList("table").indexOf(user.getIdLong()))));
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/draft/RevealPrismaTeamCommand.kt
package de.tectoast.emolga.commands.draft

import de.tectoast.emolga.buttons.buttonsaves.PrismaTeam
import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.Constants
import de.tectoast.emolga.utils.json.db
import net.dv8tion.jda.api.interactions.components.buttons.Button

object RevealPrismaTeamCommand :
    Command("revealprismateam", "Revealt die Prisma Teams lol", CommandCategory.Draft, Constants.G.FLP) {
    init {
        setCustomPermissions(PermissionPreset.fromIDs(Constants.HENNY))
        argumentTemplate = ArgumentManagerTemplate.builder()
            .add("user", "User", "Der User lol", ArgumentManagerTemplate.DiscordType.USER)
            .setExample("!revealprismateam @HennyHahn")
            .build()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val user = e.arguments.getMember("user")
        val prisma = db.league("Prisma")
        val picks = prisma.picks
        val jsonList = picks[user.idLong]!!.reversed()
        e.textChannel.sendMessage("**" + user.effectiveName + "**")
            .setActionRow(Button.primary("prisma;lol", "Nächstes Pokemon"))
            .queue { message ->
                prismaTeam[message.idLong] = PrismaTeam(
                    jsonList.map { it.name }, prisma.table.indexOf(user.idLong)
                )
            }
    }
}
